src/gcp/artifactregistry.ts

```diff
@@ -21,8 +39,71 @@ export interface Operation {
   };
 }
 
+export interface CleanupPolicyCondition {
+  tagState: string;
+  olderThan: string;
+  packageNamePrefixes?: string[];
+  tagPrefixes?: string[];
+  versionNamePrefixes?: string[];
+  newerThan?: string;
+}
+
+export interface CleanupPolicyMostRecentVersions {
+  packageNamePrefixes?: string[];
+  keepCount: number;
+}
+
+export interface CleanupPolicy {
+  id: string;
+  action: string;
+  condition?: CleanupPolicyCondition;
+  mostRecentVersions?: CleanupPolicyMostRecentVersions;
+}
+
 /** Delete a package. */
 export async function deletePackage(name: string): Promise<Operation> {
   const res = await client.delete<Operation>(name);
   return res.body;
 }
+
+/**
+ * Get a repository from Artifact Registry.
+ */
+export async function getRepository(repoPath: string): Promise<Repository> {
+  const res = await client.get<Repository>(repoPath);
+  return res.body;
+}
+
+/**
+ * Update an Artifact Registry repository.
+ */
+export async function updateRepository(repo: Partial<Repository>): Promise<Repository> {
+  const updateMask = proto.fieldMasks(repo, "cleanupPolicies", "labels");
+  if (updateMask.length === 0) {
+    const res = await client.get<Repository>(repo.name!);
+    return res.body;
+  }
+  const res = await client.patch<Partial<Repository>, Repository>(`/${repo.name}`, repo, {
+    queryParams: { updateMask: updateMask.join(",") },
+  });
+  return res.body;
+}
+
+export type Permissions = { permissions: string[] };
+
+/**
+ * Test IAM permissions for an Artifact Registry resource.
+ *
```

Comment:
This doesn't seem to be implemented anywhere; it may be worth yanking. testIamPermissions is meant for UX design and will fail open (suggesting the user has permissions they don't actually ahve)

Reply:
Removed.